### DISCUSSION TITLE
Max number of nodes/edges?

### DISCUSSION THREAD

**Discussion by ngbrenda** [arun1729/cog#49] [Q&A]
In the [intro](https://cogdb.io/intro/), it says that it's limited to "low" workloads. Do we have any concrete numbers on the max nodes/edges/triples that it supports? Thanks.

**Comment by arun1729** [comment]
Good question! Here are some numbers from my testing:


| Metric | Value |
| :--- | :--- |
| **Write Speed (Batch)** | ~4,400 edges/sec |
| **Write Speed (Individual)** | ~2,700 edges/sec |
| **Read Speed** | 20,000+ ops/sec |

### Tested Graph Sizes
* **Social Network Graphs:** ~12,500 edges
* **Vector Embeddings:** Up to 50,000
* **Practical Sweet Spot:** 10,000 – 100,000 triples

You could probably go further depending on the hardware you are running.

Not designed for Million+ node graphs, heavy concurrent writes, or large scale workloads.

To test performance on your own hardware, run:
```bash
python3 test/benchmark.py
```
